src/kit/internal/theme.ts

```diff
@@ -0,0 +1,304 @@
+/* eslint-disable sort-keys-fix/sort-keys-fix */
+
+import React, { useContext } from 'react';
+
+import { isColor, rgba2str, rgbaMix, str2rgba } from 'kit/internal/color';
+import { ValueOf } from 'kit/utils/types';
+
+const STRONG_WEAK_DELTA = 45;
+
+export type Mode = ValueOf<typeof Mode>;
+
+export const Mode = {
+  System: 'system',
+  Light: 'light',
+  Dark: 'dark',
+} as const;
+
+export type Theme = Record<keyof typeof themeBase, string>;
+
+export const MATCH_MEDIA_SCHEME_DARK = '(prefers-color-scheme: dark)';
+export const MATCH_MEDIA_SCHEME_LIGHT = '(prefers-color-scheme: light)';
+
+interface ThemeState {
+  themeIsDark: boolean;
+  theme: Theme;
+}
+
+export const ThemeContext = React.createContext<ThemeState | undefined>(undefined);
+
+export const useThemeState = (): { themeState: ThemeState } => {
```

Comment:
This name is wrong now.

Reply:
naming has been updated

Reply:
I think you missed the issue here which is that this isn't state.